### PR TITLE
Got limelight into a subsystem

### PR DESCRIPTION
**Pull request by mihir-jain**
Please make sure that all the features that will be needed are in there. I included all the features of limelight that I use/think will be useful, but that still leaves a lot that aren't included.

### CODE REVIEW

File: code/src/main/java/frc/robot/subsystems/Limelight.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+   * Getting limelight values
+   */
+  private NetworkTable m_limelight;
+
+  public Limelight() {
+    m_limelight = NetworkTableInstance.getDefault().getTable("limelight");
+    m_limelight.getEntry("pipeline").setNumber(0);      
+  }
+
+  public boolean ifTargetSeen(){
```
Comment (dotnetprofessional) [inline code/src/main/java/frc/robot/subsystems/Limelight.java]: I'd recommend isTargetVisible (its common to use the prefix "is" for Boolean type methods.

File: code/src/main/java/frc/robot/subsystems/Limelight.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    return m_limelight.getEntry("tv").getDouble(0.0) == 1.0;
+  }
+
+  public void LEDState(int state){
```
Comment (dotnetprofessional) [inline code/src/main/java/frc/robot/subsystems/Limelight.java]: For this method I'd suggest setLEDState so its clearer its an action.

File: code/src/main/java/frc/robot/subsystems/Limelight.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  public int getHeightAngularOffset(){
+    return (int) m_limelight.getEntry("ty").getNumber(0);
+  }
+
+  public int getRotationalOffset(){
+    return (int) m_limelight.getEntry("tx").getNumber(0);
+  }
+
+  public int getAreaOfScreenTakenByTheReflectiveTape(){
```
Comment (dotnetprofessional) [inline code/src/main/java/frc/robot/subsystems/Limelight.java]: maybe getAreaOfScreenTakenByTarget

File: code/src/main/java/frc/robot/subsystems/Limelight.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    m_limelight = NetworkTableInstance.getDefault().getTable("limelight");
+    m_limelight.getEntry("pipeline").setNumber(0);      
+  }
+
+  public boolean ifTargetSeen(){
+    return m_limelight.getEntry("tv").getDouble(0.0) == 1.0;
+  }
+
+  public void LEDState(int state){
+    m_limelight.getEntry("ledMode").setNumber(state);
```
Comment (thetaspirit) [inline code/src/main/java/frc/robot/subsystems/Limelight.java:39]: If you're going to make the user give numbers like these, at least write javadoc explaining what they mean.  What would be preferable is you can make enums that correspond to the numbers

**Review by thetaspirit — CHANGES_REQUESTED** [review]

File: code/src/main/java/frc/robot/subsystems/Limelight.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  }
+
+  public int getPipeline(){
+    return (int) m_limelight.getEntry("getpipe").getNumber(0);
```
Comment (thetaspirit) [inline code/src/main/java/frc/robot/subsystems/Limelight.java:46]: if you're going to be able to get the pipeline, you should also be able to set it.  Note that this requires for the limelight to be configured to "Ignore NetworkTables Index" which can be set through the web interface

**Review by thetaspirit — COMMENTED** [review]
For most of the getters, I'd suggest you be more detailed about what the methods return in the javadoc.  Be sure to include units

File: code/src/main/java/frc/robot/subsystems/Limelight.java (hunk 4 above)
Comment (mihir-jain) [reply code/src/main/java/frc/robot/subsystems/Limelight.java:39]: what are enums?

Comment (thetaspirit) [reply code/src/main/java/frc/robot/subsystems/Limelight.java:39]: easiest thing t o do would follow the pattern you seen on lines 24-42 [here ](https://github.com/wpilibsuite/allwpilib/blob/master/wpilibj/src/main/java/edu/wpi/first/wpilibj/XboxController.java)

**Review by thetaspirit — APPROVED** [review]

**Pull request merged**